optuna/pruners/_median.py

```diff
@@ -60,10 +60,18 @@ def objective(trial):
             Interval in number of steps between the pruning checks, offset by the warmup steps.
             If no value has been reported at the time of a pruning check, that particular check
             will be postponed until a value is reported.
+        n_min_trials:
+            Minimum number of trials to perform pruning at each step. If the number of reported
+            intermediate values in a trial is less than ``n_min_trials`` at a step,
+            the trial is not pruned at this step.
     """
 
     def __init__(
-        self, n_startup_trials: int = 5, n_warmup_steps: int = 0, interval_steps: int = 1
+        self,
+        n_startup_trials: int = 5,
+        n_warmup_steps: int = 0,
+        interval_steps: int = 1,
+        n_min_trials: int = 1,
```

Comment:
How about making the argument keyword-only? It will prevent user confusion from future extensions.
```suggestion
        *,
        n_min_trials: int = 1,
```